core/src/dird/reload.cc

```diff
@@ -183,20 +182,35 @@ bool CheckResources()
     }
   }
 
-  if (OK) {
-    CloseMsg(nullptr);              /* close temp message handler */
-    InitMsg(nullptr, me->messages); /* open daemon message handler */
-    if (me->secure_erase_cmdline) {
-      SetSecureEraseCmdline(me->secure_erase_cmdline);
-    }
-    if (me->log_timestamp_format) {
-      SetLogTimestampFormat(me->log_timestamp_format);
+  ConsoleResource* console;
+  BStringList consoles_with_auth_problems;
+  foreach_res (console, R_CONSOLE) {
+    if (console->use_pam_authentication_
+        && (console->user_acl.ACL_lists[0] || console->user_acl.profiles)) {
+      consoles_with_auth_problems.push_back(
+          std::string(console->resource_name_));
```

Comment:
With `emplace_back()` you can just pass your arguments into the constructor of the contained type (and save a copy operation):
```suggestion
consoles_with_auth_problems.emplace_back(console->resource_name_);
```